lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018 - Matthias Hollerbach <mail@matthias-hollerbach.de>
+# Copyright (c) 2016 - John Calixto <john.calixto@nordstrom.com> and Nordstrom, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: macos_pref
+author:
+    - Matthias Hollerbach (@kinglouie)
+    - John Calixto (@nordjc)
+short_description: Manipulates macOS preferences including complex data types.
+version_added: "2.6"
+description:
+    - This module allows users to create, read, update, and delete system and application preferences on macOS
+      including deeply nested dictionary values like those in the C(com.apple.finder) domain.
+    - It also provides a convenient C(merge) strategy for assignments in nested dictionaries so that users can very
+      specifically target nested keys without having to assign unchanged peer or parent values.
+requirements:
+    - Target machine should be running macOS
+    - PyObjC (preinstalled by Apple with the operating system)
+options:
+    domain:
+        description:
+            - The preference domain. E.g. C(com.apple.finder), C(/path/to/some.plist), C(NSGlobalDomain).
+        required: false
+        default: NSGlobalDomain
+    host:
```

Comment:
Is this some kind of ability to administer multiple machines from same Mac? Maybe mention how this differs from the host Ansible is running on (not sure how common is knowledge about fleet management amongst somewhat regular Mac users, I guess).

Reply:
This is not about multiple machines
The "default" for preferences is `anyHost`. Using the bundle name com.example for an example, these preferences would be stored at

```~/Library/Preferences/com.example.plist```
What makes this "anyHost" is the assumption that your home directory is stored on a network share. There is no additional mechanism to synchronize it across computers. Generally speaking, you should always use anyHost unless you are setting a value which really does need to be specific to the physical machine the user is sitting at.

Preferences which are set for the current host are stored under the ByHost directory, with a file name that includes a UUID for the computer, such as:

```~/Library/Preferences/ByHost/com.example.74263C97-28D5-4AF0-8E7A-3169F1CCB545.plist```

Reply:
How about mentioning something along the lines of "Unless you know what you are doing, keep `anysoft` as value" and/or maybe linking to/mentioning some official Apple docs on the topic?

Reply:
Will update 👍